### DISCUSSION TITLE
Slash

### DISCUSSION THREAD

**Discussion by BV-Labs** [rern/rAudio#1270] [Bug Reports]
As others have noted I cannot type either a \ or /  into the field for "Share name" so I can't put the files path to the music library. 
This is VERY annoying and makes it useless to connect to the NAS. 
Whether we are doing a clean install or upgrading an existing version of rAudio, this error remains.
Is it possible to fix this error? Many users will be very grateful.

With respect to the work of developers, your loyal user. :)

![SlashError](https://user-images.githubusercontent.com/76734702/225599048-78278f8f-969b-42a2-94ef-e7f28c2b1f14.jpg)

**Comment by R101G** [comment]
No, you do not need to enter anything other than the IP and the share name. So long as the share name you enter has a matching [Share] entry configured in the NAS, it works. You just have to make sure the path set in the NAS smb.conf is valid. If the NAS is a router, you may need vers=2.0 in the options field.

**Reply by BV-Labs** [reply]
Hmmm... How do I set the path to a specific folder on my NAS? Here is an example path to my music on NAS: **NAS/Music/MyMusic**
If you don't understand: I can't enter the character / - it immediately deletes itself.

My NAS - Synology

**Comment by R101G** [comment]
I have previously made the same mistake myself, probably an assumption carried over from the old Rune setup.

How you set up the share will depend on the NAS. If Samba is running on the NAS, it just looks for a share name which matches one it has in it's configuration. If you want to go straight to MyMusic, you simply set up a share name [MyMusic] in the NAS with a path NAS/Music/MyMusic. If it's a router, you might only need /Music/MyMusic depending on the setup method. Without knowing what kind of NAS you have, it's a bit difficult to help. Best way is to test with a file manager after setting up the NAS share. Enter smb://10.0.1.1/MyMusic and see if it works from there. If it does, entering MyMusic as the share name in rAudio should work too.

**Reply by BV-Labs** [reply]
You don't seem to understand me.
Okay, let's try it differently: how can I write this text in "Share name" - **Music/MyMusic** ?
Or even simpler - how do I just type the character / in this line?

**Reply by R101G** [reply]
You do not need to. Share name is the name of the share which Samba looks for. It does not need the path - that is set in the NAS configuration, which sharing a folder sets up automatically.

**Comment by R101G** [comment]
I had a quick look at Synology products. It looks as if there should be no problem if you have a shared folder called MyMusic on it.

**Reply by BV-Labs** [reply]
You didn't answer my question: how do I enter this character - / ?
It doesn't matter what NAS I have.

**Reply by R101G** [reply]
You can't. There is no reason to.

**Reply by BV-Labs** [reply]
Is this an artificial limitation?
For what? Why? Remove then generally the possibility of connecting the network drive. I'm shocked.

**Reply by R101G** [reply]
That is just how it works. Share the folder on the NAS, and use the folder name as the share name. Samba should find it.

**Reply by BV-Labs** [reply]
```js
			$share.on( 'keyup paste', function() {
				setTimeout( () => {
					var slash = $( '#infoContent input[type=radio]:checked' ).val() === 'cifs' ? /[\/\\]/g : /\\/g;
					$share.val( $share.val().replace( slash, '' ) );
				}, 0 );
			} );
```
Very elegant solution. Only one question - for **what**?

**Reply by rern** [reply]
@BV-Labs,

Please try editing the line at `/[\/\\]/g` to `/^[\/\\]/g`

**Comment by EarsofTin** [comment]
What should the original poster do if his NAS if not a Samba server? I too cannot enter the path name containing "/" and I'm pretty sure I was able to do so in the past.  For what it's worth, this is what my mounts look like:

# cat /etc/fstab
PARTUUID=922682e3-01  /boot  vfat  defaults,noatime  0  0
PARTUUID=922682e3-02  /      ext4  defaults,noatime  0  0
//192.168.1.4/Public/Shared\040Music/New  /mnt/MPD/NAS/New  cifs  noauto,username=guest,uid=45,gid=45,iocharset=utf8  0  0
//192.168.1.4/Public/Shared\040Music/Main  /mnt/MPD/NAS/Main  cifs  noauto,username=guest,uid=45,gid=45,iocharset=utf8  0  0

**Reply by R101G** [reply]
Does this work?

https://www.howtogeek.com/176471/how-to-share-files-between-windows-and-linux/

So long as you can mount the smb share with just an IP and a folder name, you should not need the slash symbol for the rAudio setup. I don't have any Windows machines to try it on.

**Comment by EarsofTin** [comment]
Unfortunately for me, the NAS I use (WD MyCloud) has some peculiarities that creates a difficulty.  The NAS provides a permanently shared folder "Public" that is further divided into Music, Photos and Videos. Subdirectories can be then be created in the "Public" share such as MomsMusic, DadsMusic, XmasMusic etc.

My problem is how to add just DadsMusic (about 70k tracks) to rAudio without adding everthing in Public (over 200k tracks). I desperately want to avoid migrating 4TB of music to a private folder and than creating a share to it.

For many years (going back to Rune+Rern enhancements) I simply entered the directory path including slashes as the share name.  That no longer works. 

I tried using ssh to add another directory path using the current shares as a template with no success. Note that the current shares include slashes. Probably additional code other than /etc/fstab needs to be edited.

With LMS I add the entire Public share, then specify only the subdirectories that I want scanned and added to the library.  Perhaps a similar approach would work in rAudio but I don't know how to separate mounting a share from scanning the same share.

In the meantime my current shares including slashes are "grandfathered" and I make, remove and edit subdirectories inside of them.  So I'll get by.

**Reply by cmh714** [reply]
I have this working with my 2 WD NAS drives...but I do use NFS, would think smb/cifs should be similar
screenshot from the system page:
![image](https://user-images.githubusercontent.com/8241076/226113888-249968a9-3407-4cfb-ad27-56cd530b11e4.png)

fstab:
192.168.2.131:/nfs/Public/Shared\040Music/Live  /mnt/MPD/NAS/Live\040Dead\040&\040Jerry  nfs   defaults,noauto,bg,soft,timeo=5,ro,auto  0  0
192.168.2.144:/nfs/Public/Shared\040Music       /mnt/MPD/NAS/Other\040Music              nfs   defaults,noauto,bg,soft,timeo=5,rw,auto  0  0
192.168.2.144:/nfs/MyPassport/Music             /mnt/MPD/NAS/Passport                    nfs   defaults,noauto,bg,soft,timeo=5,ro,auto  0  0

**Reply by EarsofTin** [reply]
I posted the contents of my fstab above and tried to clone it to add a new subdirectory but no success.  Thanks for reminding me that there are only three types of music in the world; the Grateful Dead, other stuff I like and everything else.  Part of my problem is that my Dead collection overwhelms rAudio so I have to play it via VLC.

**Reply by cmh714** [reply]
not sure why the problems....thats more than 6TB of flac files, many HD and over 117k of songs...but I wish you luck!!

**Reply by cmh714** [reply]
maybe a switch to nfs is what you need......pretty sure nfs is better than cifs/smb for streaming. YMMV :)

**Reply by cmh714** [reply]
Ok, yeah so I see that the CIFS/SMB field doesnt allow either slash, forward or backward....IMHO, it takes a few minutes to get NFS setup and it works. And the settings field is allowing the slashes...again, just my .02

**Comment by ceejayemm** [comment]
Might be worth being consistent with your '/' and '\' - I don't know if it matters but ....

@EarsofTin  - I know you don't want to move your existing Public folder structure to individual shares but it might give you a simpler share structure that rAudio accepts as standard.  Depending on how your 'NAS' defines CIFS/Samba shares you may not necessarily need to physically move your existing files just redefine your shares to represent the shares you need .

Windows does let you share a upper level folder as say \\192.168.86.50\Music, this could be mounted in rAudio as:

Server IP: 192.168.86.50
Share Name: Music

but then in Windows you could also connect to the same share AND a lower lever folder as in:

\\192.168.86.50\Music\Classical

and again as:

\\192.168.86.50\Music\Classical\Elgar

both are valid in Windows - but may not be in Linux CIFS hence rAudio trying to restrict the use of the '\' in the lover level access.  Therefore split up the Music folder into its component parts by sharing

Music                           as      Music
Music\Classical            as      Classical
Music\Classical\Elgar   as      Elgar

all of which will be valid in rAudio.

Just my twopeneth for what its worth

Chris
